File: JavaScript/Source/Histogram.js
export class Histogram {

  /** @param s {string}
   * @returns string */
  static hist(s) {
    const count = {};
    for (const letter of s)
      count[letter] = count[letter] ? count[letter] + 1 : 1;

    let res = [];
    for (const letter of 'uwxz') {
      if (count[letter]) {
        const num = `${count[letter]}      `.substring(0, 6);
        const stars = '*'.repeat(count[letter]);
        res.push(`${letter}  ${num}${stars}`);
      }
    }

    return res.join('\n');
  }
}

File: Nim/Source/Histogram.nim
import std/tables
import std/strutils

type Histogram = ref object of RootObj

proc hist*(s: string): string =
	
	var count = initTable[char, int]()
	for letter in s:

		count[letter] = if hasKey(count, letter): count[letter] + 1 else: 1

	res: seq[string]
	for letter in "uwxz":
	if hasKey(count, letter) and count[letter] > 0:

		let num = alignLeft(intToStr(count[letter]), 6)
		let stars = '*'.repeat(count[letter])
		res.add(letter & "  " & num & stars)

	return join(res, "\n")
